### ISSUE TITLE
VMS: How do we determine if a WP participant is paid or not

### ISSUE THREAD

**Issue by WaBirder57** [washingtontrails/vms#332]
@mscholtz @kuffelman We need to track if a work party participant is paid or not for each particular work party.  While we track the Contact Crew Leader Paid status to use as a default for Crew Leaders, other participants for a particular work party may be paid (such as staff filling in as an ACL) or other person.  Also, a Paid Crew leader might participate as a volunteer, so they are not always paid.

We need to track this information for our grant reports which tally up Volunteer Hours. Also we need to expense the cost of paid WP participants.  How do we record this information for each WP Registration?

**Comment by mscholtz** [comment]
@WaBirder57 @kuffelman Do you already have a way to indicate who's WTA staff on the contact record?  If so, that might take care of reporting on staff participation.  Would that be good enough?  In what situations are non-staff non-CLs paid to participate?

When you say you need to "expense the cost", is that something that needs to be tracked in SF?  Or elsewhere (like in your books)?

This would ideally be a new story since this would be a new feature, but let's talk it out here to determine if we really need something new.

**Comment by WaBirder57** [comment]
@ebskopin We do have the Staff Status field on the contact. This is a pick list with values of Current, Past, Prospective.  Know that there are some WP where staff participate as Paid or another WP as a Volunteer.  Thus, regardless of their current "staff status" I think we need to track at the WP registration level if someone is paid or volunteer.  If we have a flag on the WP Registration as paid or not we could use it for anyone on a WP.

Yes, the Expense the cost is going to go in a new story that has to do with the matching of WP work to our various grants.  We had talked a little about this and you suggested some discovery time.  We don't need this to launch but we do need to keep the detail records that will be needed such as if a WP participant was paid for that particular WP.

Let me know what you think.

**Comment by mscholtz** [comment]
@WaBirder57 we could have a flag on the registration, but that's not how we're doing it for CLs.  For CLs we're flagging it on the contact and assuming they're either always paid or always unpaid.  I'm a little hesitant to have two very different methods of tracking this.  

I guess one answer could be have a Paid flag on the registration, and default that to checked for CLs that have the paid flag on the contact.  And otherwise default to unchecked.  Would that work?  

Of course you guys would have to be careful about checking it if you want the reports to be accurate.

Still seems like a (small) story to me.

**Comment by ebskopin** [comment]
From my experience, the majority of times staff attend WPs they are considered "paid." It might be best to treat all current staff as we do CLs in the system. Would this mean that there would be an option to un-check "paid" on the registration?

**Comment by WaBirder57** [comment]
I reviewed this with Rebecca, and while the fields on the Contact record (CL Paid or Staff Status) can help set the default for whether an individual is paid on a specific work party we need the ability to change this at the registration level.  The trail team will review this before and after the work party to adjust if needed.  

For example we might a have a monthly report showing WP Participation with Paid and Volunteer statistics.  We could well find that we need to adjust the "paid" status for an individual in a particular WP.

Here is one example where the default would need to be manually over written, typically WTA Staff who are in the Trail Program (eg Rebecca or Elliott) participate in a WP they would be considered paid, this is part of their job.  If Kate from the development department was on a WP should would typically be considered a Volunteer, WP is not part of her job.

**Comment by mscholtz** [comment]
OK, got it.
Charlie, do you want to write a couple of stories for this?  One for the
paid/unpaid status, and one to track expenses etc?

On Fri, Oct 30, 2015 at 10:43 AM, Charles Kahle <notifications@github.com>
wrote:

> I reviewed this with Rebecca, and while the fields on the Contact record
> (CL Paid or Staff Status) can help set the default for whether an
> individual is paid on a specific work party we need the ability to change
> this at the registration level. The trail team will review this before and
> after the work party to adjust if needed.
>
> For example we might a have a monthly report showing WP Participation with
> Paid and Volunteer statistics. We could well find that we need to adjust
> the "paid" status for an individual in a particular WP.
>
> Here is one example where the default would need to be manually over
> written, typically WTA Staff who are in the Trail Program (eg Rebecca or
> Elliott) participate in a WP they would be considered paid, this is part of
> their job. If Kate from the development department was on a WP should would
> typically be considered a Volunteer, WP is not part of her job.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/washingtontrails/vms/issues/332#issuecomment-152598469>
> .
>



-- 

Matthew Scholtz
Percolator Consulting
(206) 289-0816 x3

http://percolatorconsulting.com/

**Comment by WaBirder57** [comment]
I have created story #322 in the back log: Work Party registration tracks if participant is volunteer or paid.

 

Matthew, help me understand what needs to go into the “Track Expenses” story.  I think this needs more discovery to have better clarity on this.

 

A little of the back story is we get a number of grants to help pay for our work parties.  These grants are typically tied to specific types of lands, eg: USFS Lands, National Park Lands, Trails in the Columbia Gorge, King County, etc.  We allocated volunteer effort to fulfill obligations of the grants and expenses related to the work (mileage, food, staff hours, etc).  Often specific work party (expenses and volunteer time) can be attributed to one or more grants.  Also, it might only be at the end of the season that we can look at the combination of Grants and Work Parties to figure out the optimal allocation between the various grants.

 

Charlie
